service/src/ocall_bridge/sidechain_ocall.rs

```diff
@@ -112,4 +129,118 @@ where
 
 		status
 	}
+
+	fn fetch_sidechain_blocks_from_peer(
+		&self,
+		last_known_block_hash_encoded: Vec<u8>,
+		shard_identifier_encoded: Vec<u8>,
+	) -> OCallBridgeResult<Vec<u8>> {
+		let last_known_block_hash: BlockHash =
+			Decode::decode(&mut last_known_block_hash_encoded.as_slice()).map_err(|_| {
+				OCallBridgeError::FetchSidechainBlocksFromPeer(
+					"Failed to decode last known block hash".to_string(),
+				)
+			})?;
+
+		let shard_identifier: ShardIdentifier =
+			Decode::decode(&mut shard_identifier_encoded.as_slice()).map_err(|_| {
+				OCallBridgeError::FetchSidechainBlocksFromPeer(
+					"Failed to decode shard identifier".to_string(),
+				)
+			})?;
+
+		let tokio_handle = self.tokio_handle.get_handle();
+
+		let signed_sidechain_blocks = tokio_handle
+			.block_on(
+				self.peer_block_fetcher
+					.fetch_blocks_from_peer(last_known_block_hash, shard_identifier),
+			)
+			.map_err(|e| {
+				OCallBridgeError::FetchSidechainBlocksFromPeer(format!(
+					"Failed to execute block fetching from peer: {:?}",
+					e
+				))
+			})?;
+
+		Ok(signed_sidechain_blocks.encode())
+	}
```

Comment:
This is the implementation that calls into the peer block fetcher from the sidechain crate. Because it's an asynchronous call that we run synchronously, we need a tokio handle.